File: python/fizzbuzz/src/fizzbuzz.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

from __future__ import unicode_literals

class FizzBuzz:

	def convert(self, number):

		# throws an exception if can convert
		number = int(number)

		result = ""

		if number % 3 == 0:
			result = "fizz"
		if number % 5 == 0:
			result += "buzz"
		if result == "":
			result = str(number)

		return result

	def display(self):
		for x in range(1, 101):
			print self.convert(x)

File: php/fizzbuzz/src/FizzBuzz.php
<?php

namespace Kata;

class FizzBuzz
{
    /**
     * @param  int  $number
     * @return string
     */
    public function convert($number)
    {
        $result = '';

        if ( $this->isNotDivisibleBy(3, $number) AND $this->isNotDivisibleBy(5, $number)) {
            return (string)$number;
        }

        if ($this->isDivisibleBy(3, $number)) {
            $result .= 'fizz';
        }
        if ($this->isDivisibleBy(5, $number)) {
            $result .= 'buzz';
        }

        return $result;
    }

    /**
     * @param  int  $dividend
     * @param  int  $number
     * @return boolean
     */
    public function isDivisibleBy($dividend, $number)
    {
        return $number % $dividend === 0;
    }

    /**
     * @param  int  $dividend
     * @param  int  $number
     * @return boolean
     */
    public function isNotDivisibleBy($dividend, $number)
    {
        return ! $this->isDivisibleBy($dividend, $number);
    }

    /**
     * @return string
     */
    public function display()
    {
        foreach (range(1, 100) as $number) {
            echo $this->convert($number) . PHP_EOL;
        }
    }
}
